google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -476,6 +477,18 @@ static String getServiceAccountProjectId(String credentialsPath) {
     return project;
   }
 
+  /* 
+   * Returns true if the projectId is valid.
+   * projectId must be between 6 and 30 characters
+   * projectId can have lowercase letters, digits or hyphens
+   * and must start with a lowercase letter
```

Comment:
I think we should add a reference for this. Is there a web page where this comes from? So we can easily look it up if the requirement ever changes.